From 8c111f21aded9342d409c63ad0d305dea2074733 Mon Sep 17 00:00:00 2001
Message-Id: <8c111f21aded9342d409c63ad0d305dea2074733.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Rafael Barbalho <rafael.barbalho@intel.com>
Date: Wed, 13 Aug 2014 20:56:46 +0100
Subject: [PATCH 243/312] FOR_UPSTREAM [VPG]: drm/i915: Fix crash when waiting
 for flip notification

We don't always have a previous framebuffer object when waiting for a flip
notification.

Issue found while testing clone mode on the cherrytrail rvp.

Issue: GMIN-3129
Change-Id: I679bd8d502bdb40eae7783fb3928a90834a58c06
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 6a1e968..74c50cc 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -3857,8 +3857,10 @@ void intel_crtc_wait_for_pending_flips(struct drm_crtc *crtc)
 			intel_unpin_sprite_work_fn(
 				&intel_crtc->sprite_unpin_work->work);
 			obj = intel_crtc->sprite_unpin_work->old_fb_obj;
-			atomic_clear_mask(1 << intel_crtc->plane,
-				&obj->pending_flip.counter);
+
+			if (obj)
+				atomic_clear_mask(1 << intel_crtc->plane,
+					&obj->pending_flip.counter);
 
 			spin_lock_irqsave(&dev->event_lock, flags);
 			intel_crtc->sprite_unpin_work = NULL;
-- 
1.7.9.5

